src/Orchard.Environment.Shell/ShellStateUpdater.cs

```diff
@@ -52,7 +53,11 @@ public class ShellStateUpdater : IShellStateUpdater
             var shellState = await _stateManager.GetShellStateAsync();
 
             // start with description of all declared features in order - order preserved with all merging
-            var orderedFeatureDescriptors = _extensionManager.GetExtensions().Features;
+            var allUnorderedFeatures = _extensionManager.GetExtensions().Features.ToArray();
```

Comment:
Extension Manager doesnt know about a shell... But I wonder if we can put it else where. Let me have a think.

Load Feature needs to stay on ExtensionManager as that's much lower level